src/context.rs

```diff
@@ -514,14 +530,239 @@ pub fn is_statement_read_only(statement: &DFStatement) -> bool {
     }
 }
 
+// Appropriated from https://github.com/delta-io/delta-rs/pull/1176; once the DELETE and UPDATE ops
```

Comment:
I'd move these to something like `/src/delta-rs/backports.rs` like we have with some of datafusion's parser code to make it explicit this is temporary code and to avoid ballooning this file's line length.

Reply:
Good point, moving.